source/common/quic/quic_filter_manager_connection_impl.h

```diff
@@ -59,11 +60,63 @@ class QuicFilterManagerConnectionImpl : public Network::ConnectionImplBase,
   void readDisable(bool /*disable*/) override { ASSERT(false); }
   void detectEarlyCloseWhenReadDisabled(bool /*value*/) override { ASSERT(false); }
   bool readEnabled() const override { return true; }
+
+  // TODO(alyssawilk, danzh), sort out which of these need to be handled
+  // locally.
+  class ConnectionInfoProviderShim : public Network::ConnectionInfoSetter {
```

Comment:
Sorry, I wasn't able to figure out why we need this shim. Doesn't `network_connection_->connectionSocket()->connectionInfoProvider().setSslConnection(quic_ssl_info_)` work? If not, a comment about this wrapper class will be very appreciated.

Reply:
ah, I thought quic connections shared the socket object along with the kernel socket.  I feel much better about this PR if I can get rid of this mess.